flink-yarn/src/test/java/org/apache/flink/yarn/YarnResourceManagerTest.java

```diff
@@ -304,19 +318,149 @@ public void grantLeadership() throws Exception {
 		/**
 		 * Start the resource manager and grant leadership to it.
 		 */
-		public void startResourceManager() throws Exception {
+		void startResourceManager() throws Exception {
 			resourceManager.start();
 			rmServices.grantLeadership();
 		}
 
 		/**
 		 * Stop the Akka actor system.
 		 */
-		public void stopResourceManager() throws Exception {
+		void stopResourceManager() throws Exception {
 			rpcService.stopService().get();
 		}
 	}
 
+	static class TestingContainer extends Container {
```

Comment:
I am not convinced about this. The mock was introduced with [FLINK-8177](https://issues.apache.org/jira/browse/FLINK-8177) because Hadoop 2.9 added new methods.